### ISSUE TITLE
Unable to parse `allow-insecure-host` in pyproject.toml

### ISSUE THREAD

**Issue by frague59** [astral-sh/uv#6741]
# pyproject.toml file

```sh
$ cat pyproject.toml
...
[tool.uv]
allow-insecure-host = ["https://example.com"]  # I've tried "example.com" too...

[tool.uv.pip]
extra-index-url = [
    "https://pypi.org/simple/",
    "https://example.com/simple/",  
]
```
# `uv pip install`

```sh
$ uv pip install -r requirements.txt  # some packages are in the example.com server
warning: Failed to parse `pyproject.toml` during settings discovery:
  TOML parse error at line 109, column 1
      |
  109 | [tool.uv]
      | ^^^^^^^^^
  invalid type: string "https://example.com", expected struct TrustedHost
```

**It seams the config de-serializer does not read the `allow-insecure-host` parameter ["https://example.com"] (resp. ["example.com"]) correctly**

The `--trusted-host` parameter works fine (when removing the `[tool.uv]` section from `pyproject.toml`.

**Comment by frague59** [comment]
It works with :
```toml

[tool.uv]
allow-insecure-host = [{ host = "example.com" }]
```

**Comment by frague59** [comment]
# BTW :

```sh
$ uv --version 
uv 0.3.5
```

**Comment by charliermarsh** [comment]
Apologies, this was fixed in #6716 which isn't released yet (but will be out today).
